setup/index.ts

```diff
@@ -1,11 +1,31 @@
-import { askSessionToken } from './prompt';
-import { get as getEnvValue, set as setEnvValue } from './env';
+import {
+  /* askSessionToken, */
+
+  askIOBackendHost,
+  askIOBackendBasePath
+} from "./prompt";
+import {
+  get as getEnvValue,
+  setAll as setAllEnvValues,
+  EnvKey,
+} from "../lib/env";
+
+const ask = (key: EnvKey, promptFn: () => Promise<string>) => {
+  return getEnvValue(key).fold(promptFn, value => () =>
+    Promise.resolve(value)
```

Comment:
return TaskEither not promises

Reply:
I think I need some more expertise in `fp-ts` to appreciate the full value of this.

Thanks 💪

Reply:
https://twitter.com/gabro27/status/1085105462096343041

summary: taskeither are lazy and have the same semantics as the other monadic structures (ie. eithers): you can seamless chain them, mapping, folding, etc.. you don't need special constructures (ie. async / await) to work with them since these structures (monads) share the same interface.

there's a very good blog series by gcanti here:
https://dev.to/gcanti/getting-started-with-fp-ts-io-36p6